app/models/adapters/active_admin_pundit_adapter.rb

```diff
@@ -0,0 +1,46 @@
+module Adapters
+  class ActiveAdminPunditAdapter < ActiveAdmin::PunditAdapter
+    def get_policy(subject, user, resource)
+      "ActiveAdmin::#{subject}Policy".constantize.new(user, resource)
+    end
+
+    def retrieve_policy(subject)
+      case subject
+      when nil then get_policy(subject, user, resource)
+      when Class then get_policy(subject, user, subject.new)
+      else
+        return Pundit.policy!(user, subject) if active_admin_namespace?(subject)
+
+        get_policy(subject.class, user, subject)
+      end
+    end
+
+    def scope_collection(collection, _action = Auth::READ)
+      return collection if valid_collection_class?(collection)
+
+      namespaced_policy(collection).new(user, collection).resolve
+    rescue Pundit::NotDefinedError => e
+      raise e unless policy_has_scope?(default_policy_class)
```

Comment:
This seems like a usual case doesn't it? Maybe we could check before instead of using exceptions as flow control

Reply:
this here actually raises an exception, because I tried to make it similar to what [ActiveAdmin::PunditAdapter does](https://github.com/activeadmin/activeadmin/blob/93175ed7ff09d113cb77f5e946ee8f054c438775/lib/active_admin/pundit_adapter.rb#L20).